### ISSUE TITLE
Error information not printed for errors thrown while evaluating using vm.runInContext

### ISSUE THREAD

**Issue by platy** [mochajs/mocha#1658]
It looks like the runtime formats the error message in a different way for vm.runInContext and mocha seems to have trouble and doesn't print the message at all.

Test:
```javascript
var vm = require('vm');
require('should');

describe("BDD framework", function(){
    it("throws an undefined error", function() {
        //(function() {
            undefined();
        //}).should.throw(/undefined is not a function/);
    });

    it("throws an undefined error from a runInContext", function() {
        //(function() {
            vm.runInContext('undefined();', vm.createContext({}), {showErrors: true});
        //}).should.throw(/undefined is not a function/);
    });
});
```

Mocha output:
```
  0 passing (19ms)
  2 failing

  1) BDD framework throws an undefined error:
     TypeError: undefined is not a function
      at Context.<anonymous> (spec/mocha-problem-spec.js:7:13)

  2) BDD framework throws an undefined error from a runInContext:
     Error: the error {} was thrown, throw an Error :)
```

Jasmine output:
```
Failures:
1) BDD framework throws an undefined error
  Message:
    TypeError: undefined is not a function
  Stack:
    TypeError: undefined is not a function
        at Object.<anonymous> (/Users/mikebush/Workspace/spec-verify/spec/mocha-problem-spec.js:7:13)

2) BDD framework throws an undefined error from a runInContext
  Message:
    TypeError: evalmachine.<anonymous>:1
    undefined();
    ^
    undefined is not a function
  Stack:
    evalmachine.<anonymous>:1
    undefined();
    ^
    TypeError: undefined is not a function
        at evalmachine.<anonymous>:1:1
        at Object.exports.runInContext (vm.js:64:17)
        at Object.<anonymous> (/Users/mikebush/Workspace/spec-verify/spec/mocha-problem-spec.js:13:16)
```

**Comment by boneskull** [comment]
looks like `vm.runInContext` won't throw an actual `Error` for reasons unknown to me.  if anyone has any further information on why (or if this is some sort of bug), I'd like to know...

in the meantime:

```js
try {
  vm.runInContext()
} catch (e) {
  throw new Error(e);
}
```

**Comment by platy** [comment]
Thanks, I missed this reply up until now.

It looks like the way runInContext throws is fixed in v4.0.0 https://github.com/nodejs/node/pull/2108

**Comment by danielstjules** [comment]
This was likely fixed as part of https://github.com/mochajs/mocha/pull/1758 as well :)
